File: WebApplication/src/modules/student/models/education-select.dto.ts
import { AutoMap } from "@automapper/classes";
import { NomenclatureDto } from "src/infrastructure/models/nomenclature.dto";
import { CourseType } from "src/modules/application/application-one/enums/course-type.enum";
import { YearType } from "src/modules/application/application-one/enums/year-type.enum";

export class EducationSelectDto {
  @AutoMap()
  personStudentDoctoralId: number;
  @AutoMap()
  institution: NomenclatureDto;
  @AutoMap()
  researchArea: NomenclatureDto;
  @AutoMap()
  speciality: NomenclatureDto;
  @AutoMap()
  educationFormType: NomenclatureDto;
  @AutoMap()
  educationalQualification: NomenclatureDto;
  @AutoMap()
  studentCourse: CourseType | null;
  @AutoMap()
  doctoralYear: YearType | null;
  @AutoMap()
  facultyNumber: string;
  @AutoMap()
  nationality: NomenclatureDto;
}

File: StudentCredit.Application/Students/Dtos/EducationSelectDto.cs
using StudentCredit.Data.Rdpzsd.Enums;
using StudentCredit.Data.Rdpzsd.Nomenclatures;
using StudentCredit.Infrastructure.Helpers.Dtos;

namespace StudentCredit.Application.Students.Dtos
{
	public class EducationSelectDto
	{
		public int PersonStudentDoctoralId { get; set; }
		public NomenclatureDto Institution { get; set; }
		public NomenclatureDto ResearchArea { get; set; }
		public NomenclatureDto Speciality { get; set; }
		public NomenclatureDto EducationFormType { get; set; }
		public NomenclatureDto EducationalQualification { get; set; }
		public CourseType? StudentCourse { get; set; }
		public YearType? DoctoralYear { get; set; }
		public string FacultyNumber { get; set; }
		public StudentStatus StudentStatus { get; set; }
		public NomenclatureDto Nationality { get; set; }

	}
}
